### ISSUE TITLE
Enable perf testing against jlink NRE/bundle with JRE as baseline

### ISSUE THREAD

**Issue by smlambert** [AdoptOpenJDK/openjdk-tests#1933]
Related: https://github.com/AdoptOpenJDK/openjdk-build/pull/1812

As we are about to build an 'adopt_bundle' we should verify that it is performant (startup and runtime metrics).  We can also measure JRE perf and compare against it using various performance benchmarks we have in the suite.

Since the project now has some bare-metal machines for some platforms (labelled ci.role.perf), we will focus our runs on those machines/platforms.  

Initially we can run these tests in Grinders, but need to incorporate them into our overall pipeline story and configure the TRSS server (trss.adoptopenjdk.net) to monitor the runs so we can store historical data and watch for trends/regressions.

**Comment by aahlenst** [comment]
Artifacts to test for 11/x64:

* JDK: https://github.com/AdoptOpenJDK/openjdk11-binaries/releases/download/jdk-11.0.8%2B10/OpenJDK11U-jdk_x64_linux_hotspot_11.0.8_10.tar.gz
* JRE: https://github.com/AdoptOpenJDK/openjdk11-binaries/releases/download/jdk-11.0.8%2B10/OpenJDK11U-jre_x64_linux_hotspot_11.0.8_10.tar.gz
* jlink with compress=0: https://versuchsanordnung.ch/jlink-11-compress-0.tar.gz
* jlink with compress=1: https://versuchsanordnung.ch/jlink-11-compress-1.tar.gz
* jlink with compress=2: https://versuchsanordnung.ch/jlink-11-compress-2.tar.gz

Artifacts to test for 15/x64:

* JDK: https://github.com/AdoptOpenJDK/openjdk15-binaries/releases/download/jdk15-2020-08-10-09-44/OpenJDK15-jdk_x64_linux_hotspot_2020-08-10-09-44.tar.gz
* jlink with compress=0: https://versuchsanordnung.ch/jlink-15-compress-0.tar.gz
* jlink with compress=1: https://versuchsanordnung.ch/jlink-15-compress-1.tar.gz
* jlink with compress=2: https://versuchsanordnung.ch/jlink-15-compress-2.tar.gz

**Comment by smlambert** [comment]
Perf runs on [ci.role.perf&&hw.arch.x86&&sw.os.linux](https://ci.adoptopenjdk.net/label/ci.role.perf&&hw.arch.x86&&sw.os.linux) machines.  
Note: dacapo is shortened to dac and renaissance to ren.
<hr/>
**sanity.perf Initial Results for 11/x64** 

| Object Under Test    |  dac-h2   | dac-lusearch-fix| ren-akka-uct | ren-fj-kmeans | ren-future-genetic | ren-naive-bayes | ren-scala-kmeans |
| ------------- |:-------------:| -----:|-----:|-----:|-----:|-----:|-----:|
| [JDK](https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder_TKG/30/)      |   |   | | || | |
| JRE      |        |    | | || | |
| jlink compress=0 |      |    | | || | |
| jlink compress=1 |      |    | | || | |
| jlink compress=2 |      |    | | || | |

<hr/>
**sanity.perf Initial Results for 15/x64** 

| Object Under Test    |  dac-h2   | dac-lusearch-fix| ren-akka-uct | ren-fj-kmeans | ren-future-genetic | ren-naive-bayes | ren-scala-kmeans |
| ------------- |:-------------:| -----:|-----:|-----:|-----:|-----:|-----:|
| JDK      |   |   | | || | |
| JRE      |        |    | | || | |
| jlink compress=0 |      |    | | || | |
| jlink compress=1 |      |    | | || | |
| jlink compress=2 |      |    | | || | |

**Comment by smlambert** [comment]
I will plan to break this up into sections and divide it amongst GHC open-source day participants.  (fyi @llxia @sophia-guo )

**Comment by sophia-guo** [comment]
Note: jlink links in https://github.com/adoptium/aqa-tests/issues/1933#issuecomment-671942884 doesn't work anymore?

**Comment by aahlenst** [comment]
@sophia-guo Yes, I canceled the account hosting the binaries. They would not have been of much use anyway because of their age. Recreating them shouldn't be very difficult: Dump the list of modules with `java --list-modules` on the legacy JRE image and throw away what looks fishy on JRE 14 and newer. Back then, some parts started to leak into the JRE from the JDK. I'm not sure if that's still the case.
